### ISSUE TITLE
Allow users to specify the name of moving mean and variance in batch_norm API.

### ISSUE THREAD

**Issue by qingqing01** [PaddlePaddle/Paddle#8068]
The moving mean and variance are parts of models in batch norm operators. They are calculated in the training and will be used in the inference.  They are persistable variable in Fluid.  If allowing users to specify the name of moving mean and variance in batch_norm API, it's convenient to share these two arguments. It's also convenient for the model conversion by other frameworks, like https://github.com/PaddlePaddle/models/pull/617 .  

If the name is not specified in the model conversion, the conversion tool needs to generate names for parameters and other persistable variables by following the rule of name generation in Fluid. But the rules of name generation for [parameters](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/layer_helper.py#L289) and [other persistable variables](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/framework.py#L174) are ununified, it's a little complex.

In addition, the Fluid should have mechanisms to allow users to shared any persistable variable conveniently.
